libvast/include/vast/system/index.hpp

```diff
@@ -117,85 +118,19 @@ class partition_factory {
 
 private:
   filesystem_actor filesystem_;
-  const index_state& state_;
-};
-
-/// The result type of the query evaluation handler.
-enum class query_response : bool {
-  /// Signals that query evaluation is incomplete and more results can be
-  /// requested.
-  incomplete,
-  /// Signals that the query is fully evaluated.
-  complete
-};
-
-struct query_state {
-  /// The query expression.
-  vast::query query;
-
-  /// The query client.
-  receiver_actor<atom::done> client = {};
-
-  /// The number of partitions that need to be evaluated for this query.
-  uint32_t candidate_partitions = 0;
-
-  /// The number of partitions that have been reqested by the client.
-  uint32_t requested_partitions = 0;
-
-  /// The number of partitions that the query was sent to.
-  uint32_t scheduled_partitions = 0;
-
-  /// The number of partitions that are processed already.
-  uint32_t completed_partitions = 0;
-
-  /// A response promise to signal batch completion to the client.
-  caf::typed_response_promise<query_response> rp = {};
-
-  template <class Inspector>
-  friend auto inspect(Inspector& f, query_state& x) {
-    return f(caf::meta::type_name("query_state"), x.query, x.client,
-             x.candidate_partitions, x.requested_partitions,
-             x.scheduled_partitions, x.completed_partitions);
-  }
+  index_state& state_;
```

Comment:
This seems to break the usual separation of concerns. Why does the partition factory mutate the index state? Are the side effects of the partition factory documented?

Reply:
It's done so we can keep a counter for the materializations. I'll add a note.

Reply:
Can we move the materializations counter to the factory then?